### ISSUE TITLE
Option `forceAdminSSL` 301 redirects on server root not base URL 

### ISSUE THREAD

**Issue by patricktokeeffe** [TryGhost/Ghost#7467]
The config option `forceAdminSSL` creates a 301 redirect to the server root, which breaks the admin interface when Ghost is hosted in a subdirectory. 

Here is the relevant `config.js` section:

```
config = {
    production: {
        url: 'http://pi2.home/blog/',
        mail: {},
        database: {
            client: 'sqlite3',
            connection: {
                filename: path.join(__dirname, '/content/data/ghost.db')
            },
            debug: false
        },

        forceAdminSSL: true,
        server: {
            host: '127.0.0.1',
            port: '2368'
        }
    },
    ...
```

I'm using nginx for proxy and the site file looks like this:

```
server {
    listen  0.0.0.0:80;
    server_name pi2.home pi2;
    access_log /var/log/nginx/pi2.log;

    listen 443 ssl;
    ssl_certificate /etc/nginx/ssl/nginx.crt;
    ssl_certificate_key /etc/nginx/ssl/nginx.key;

    location /blog {
        # from http:// only
        #proxy_set_header X-Real-IP $remote_addr;
        #proxy_set_header HOST $http_host;
        #proxy_set_header X-NginX-Proxy true;

        # this seems to work for https://
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header Host $http_host;
        proxy_set_header X-Forwarded_Proto $scheme;

        proxy_pass http://127.0.0.1:2368;
        proxy_redirect off;
    }
}
```

The blog can be viewed at both `http://pi2.home/blog` and `https://pi2.home/blog`. However, both `http://pi2.home/blog/ghost` and `https://pi2.home/blog/ghost` redirect to `https://pi2.home/ghost` via 301 response:

![screenshot from 2016-10-01 11 47 31](https://cloud.githubusercontent.com/assets/1348834/19016382/e63c0f96-87cc-11e6-8e49-45958d9a983d.png)

Remove or comment the `forceAdminSSL: true` line and restart Ghost to restore admin interface operation. 

Xref:
- [How do I force Ghost's admin page to be server over SSL when ghost is installed in a subdirectory?](http://stackoverflow.com/q/27658522)
- kind of the opposite of #5873

---
- Ghost Version: 0.11.0
- Node Version: 2.15.10
- Browser/OS: Chromium 53/Linux Mint 18
- Database: sqlite3

**Comment by blatinier** [comment]
I experience the same kind problem on Ghost 0.11.1/Debian8/Sqlite3 just by setting the url in config.js to `https://mydomain.com/blog`.

**Comment by SeanHayes** [comment]
I have ghost setup on https://www.example.com/blog but requests to http://www.example.com/blog are being redirected to https://www.example.com/.

**Comment by patricktokeeffe** [comment]
@blatinier, @SeanHayes, this issue is specifically about the `forceAdminSSL` option causing redirects to the server root path `/`. With the config detailed in my op, I was able to run Ghost from a subdirectory. It's not immediately clear you are experiencing the same issue (though you might be).

**Comment by SeanHayes** [comment]
I think it is the same issue: https://github.com/TryGhost/Ghost/blob/02c7d039d80a9cae210d922eceee2588b7620337/core/server/middleware/check-ssl.js#L7

**Comment by SeanHayes** [comment]
I added `console.log(opts)` here: https://github.com/TryGhost/Ghost/blob/master/core/server/middleware/check-ssl.js#L16

`opts.reqUrl` is equal to `/`

**Comment by SeanHayes** [comment]
req.url was being used when we needed req.originalUrl to be used.

I think this should be fixed by PR #7548: https://github.com/TryGhost/Ghost/commit/4411f8254f90f47e4dff02963afb2131983a4462#diff-099c36a1ad7af055f336fcc377df8584R47

**Comment by SeanHayes** [comment]
I added a test in PR #7588 confirming this has been fixed.

**Comment by yavorski** [comment]
I expirience the same issue too, with Ghost 0.11.3/ubuntu_16.04.LTS/mysql just by setting url in config.js to `https://mydomain.com/blog` and not using `forceAdminSSL` variable at all.

**Comment by ErisDS** [comment]
I believe this needs fixing on the `lts` branch. It is fixed on the `master` branch but that is currently still in alpha and not planned to be released for a while yet. A PR for a fix would be welcome.

**Comment by kirrg001** [comment]
Closing this as it should be fixed in the next `LTS` release `0.11.4`.
